From bd2b941d8c62fffabab9724ae7c0a9c1476af0bf Mon Sep 17 00:00:00 2001
From: Vitaly Margolin <vmargolin@habana.ai>
Date: Tue, 17 Jun 2025 16:07:29 +0300
Subject: [PATCH] [SW-232548] lfs: add component name to start message

Add component name to start LFS logger UART message.
This is done to simplify robot testing of LFS logger
feature

Change-Id: I83ebe0894a3a58730fc248877d791a9353ce727e
Signed-off-by: Vitaly Margolin <vmargolin@habana.ai>
---

diff --git a/zephyr/lfs_log_console_sender.c b/zephyr/lfs_log_console_sender.c
index 9088e75..934a92d 100644
--- a/zephyr/lfs_log_console_sender.c
+++ b/zephyr/lfs_log_console_sender.c
@@ -71,6 +71,7 @@
 	struct k_work_q lfs_work_q;
 	k_tid_t lfs_work_q_thread_id;
 	bool time_critical;
+	const char *component_name;
 };
 
 static struct lfs_backend_data backend = {
@@ -394,7 +395,11 @@
 
 static void lfs_start_logging(void)
 {
-	LOG_INF("LFS: logger started");
+	if (backend.component_name)
+		LOG_INF("LFS: logger started by component <%s>", backend.component_name);
+	else
+		LOG_INF("LFS: logger started");
+
 	if (!backend.time_critical)
 		k_timer_start(&lfs_sync_timer, LFS_FILE_SYNC_INTERVAL, LFS_FILE_SYNC_INTERVAL);
 
@@ -444,6 +449,18 @@
 	return lfs_console_init();
 }
 
+int lfs_console_init_by_component(const char *component_name)
+{
+	backend.component_name = component_name;
+	return lfs_console_init();
+}
+
+int lfs_console_init_time_critical_by_component(const char *component_name)
+{
+	backend.component_name = component_name;
+	return lfs_console_init_time_critical();
+}
+
 k_tid_t module_lfs_setup(const char *thread_name)
 {
 	k_tid_t inactive = (k_tid_t)(-1);
diff --git a/zephyr/lfs_zephyr_init.h b/zephyr/lfs_zephyr_init.h
index 7db54fb..d2c2e52 100644
--- a/zephyr/lfs_zephyr_init.h
+++ b/zephyr/lfs_zephyr_init.h
@@ -26,6 +26,8 @@
 
 int lfs_console_init(void);
 int lfs_console_init_time_critical(void);
+int lfs_console_init_by_component(const char *component_name);
+int lfs_console_init_time_critical_by_component(const char *component_name);
 k_tid_t module_lfs_setup(const char *thread_name);
 k_tid_t module_lfs_tid_get(void);
 void module_lfs_shutdown(void);
